adding a category resource ( with most fields supported )

viggin543 opened the pull request:
added support to create flat commerce tools categories.
no support for hierarchies custom fields or assets. ( TBD )


```hsl
resource "commercetools_category" "accessories" {
  name = {
    en = "accessories"
  }
  key = "accessories"
  description = {
    en = "accessories"
  }
  slug = {
    en = "accessories"
  }

}
```

qdegraaf commented:
Hey viggin543,

Thanks for the contribution! 

On first glance the code looks good. Could you add tests for the new resource and run them to make sure everything works as intended before a more thorough review?

qdegraaf commented:
@viggin543 are you still interested in finishing this?

viggin543 commented:
Yes, sorry for the delay.

qdegraaf commented:
Hey @viggin543 thanks for adding tests.

@Ardjan-Aalberts picked this up in https://github.com/labd/terraform-provider-commercetools/pull/152 in the meantime and included the assets and hierarchies fields in a PR. This lead to an interesting potential bug in the Go-SDK, when unmarshalling the Assets response, and the change was reverted. We are still investigating this. Do you need the category resource without these fields in the short term? 

In that case we can consider merging this first and adding the other fields later, though I'd prefer merging the whole resource in functioning order if there is no rush.

viggin543 commented:
@qdegraaf yes, please.

we are using flat categories in my workplace. no hierarchies.
It also looks not trivial to implement the recursive structures, that's why I didn't do it currently.

we did publish [this](https://registry.terraform.io/providers/AdikaStyle/commercetools/latest/settings) provider that includes categories and are using it successfully for a couple of months.
So merging should be safe.



I think that

viggin543 commented:
@qdegraaf 
- added support for parent category
- added partial support for assets

can create a category with assets.
( don't support custom attributes for assets )
didn't implement updating assts yet.

Ardjan-Aalberts commented:
Hi @viggin543 we created this PR based on the code you submitted earlier: #160 . This code could not be merged before because of an issue with the mock server used for testing. This issue has been fixed. Could you let me know if #160 is complete? Thanks in advance!

viggin543 commented:
@Ardjan-Aalberts 
from what I've noticed. you missed the following:
- adding/removing assets ( you only support updating sources of an existing asset )
- referencing a parent by key ( exclusive to referencing by id )

Ardjan-Aalberts commented:
Closed, code included in #160.
